# Processes to analyze and understand structural variants

## Links to helpful sites <br> 
BLAST in python
- https://pypi.org/project/pyblastbio/
- https://www.tutorialspoint.com/biopython/biopython_overview_of_blast.htm
- https://hossainlab.github.io/PY4B/notebooks/07-DNA%20Sequence%20Statistics.html

Building a blast db locally
- https://www.ncbi.nlm.nih.gov/books/NBK569841/

Running BLASTn from command line
- https://www.ncbi.nlm.nih.gov/books/NBK569856/
- https://open.oregonstate.education/computationalbiology/chapter/command-line-blast/
- https://www.biostars.org/p/160350/

Making db and running blastn
- https://www.ncbi.nlm.nih.gov/books/NBK569856/
- https://www.metagenomics.wiki/tools/blast/blastn-output-format-6

Salmon paper on SVs
- https://www.nature.com/articles/s41467-020-18972-x

General how BLAST works and its output
- https://cs.calvin.edu/activities/blasted/other/BLAST.html
- https://resources.qiagenbioinformatics.com/manuals/clcgenomicsworkbench/650/Explanation_BLAST_output.html
- https://www.ncbi.nlm.nih.gov/books/NBK1734/
- https://guides.lib.berkeley.edu/ncbi/blast

